assets/js/components/module/ModuleApp.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * ModuleApp component.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { sprintf, __ } from '@wordpress/i18n';
+import { Fragment } from '@wordpress/element';
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { isFeatureEnabled } from '../../features';
+import Header from '../Header';
+import Alert from '../Alert';
+import PageHeader from '../PageHeader';
+import ModuleFooter from './ModuleFooter';
+import LegacyModuleApp from './LegacyModuleApp';
+import WidgetContextRenderer from '../../googlesitekit/widgets/components/WidgetContextRenderer';
+
+import DateRangeSelector from '../DateRangeSelector';
+import { CORE_MODULES } from '../../googlesitekit/modules/datastore/constants';
+
+const { useSelect } = Data;
+
+function ModuleApp( { slug } ) {
```

Comment:
This prop should be called `moduleSlug` to be a bit more verbose. We previously ran into issues where we needed multiple slugs in one component (e.g. `widgetSlug` and `moduleSlug`), and while it should be obvious here that it's a module's `slug`, being more verbose is a bit more future-proof and clear I think.

Reply:
Haha, there was some back and forth on this already (see discussion above with @tofumatt). I've changed it to `moduleSlug`.